### PR TITLE
fix(platform-browser): createElement apply attrs before attaching element to dom

### LINKED ISSUE #13947: Picture tag doesn't work if inside angular app
**I'm submitting a ...**  (check one with "x")
```
[x ] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
```

**Current behavior**
`<picture>` tag can have multiple images defined. Browser chooses which one to load based on media queries and your browser's size. An example would be:

```
<picture>
      <source media="(min-width: 1280px)"  srcset="https://placehold.it/400x400">
      <source media="(min-width: 1024px)"  srcset="https://placehold.it/300x300">
      <source media="(min-width: 640px)"  srcset="https://placehold.it/200x200">
      <source srcset="https://placehold.it/100x100">
      <img alt="" src="https://placehold.it/401x401">
</picture>
```

If `<picure>` tag is inside angular app, it always loads only the top image. `<picture>` tag outside of an angular app loads images correctly.

Warning: `<picture>` tag *works* in chrome. It doesn't work in (at least) Firefox and Safari. Don't know about IE and edge.

**Expected behavior**
Should load images correctly.

**Minimal reproduction of the problem with instructions**
https://plnkr.co/edit/qpokJtZi6U3jP864rO9X?p=preview

Try opening this link in Firefox and then resize your window. `<picture>` tag that is not in angular app works correctly and the one that is in app doesn't. They are the same tags, copy-pasted.

**What is the motivation / use case for changing the behavior?**
We'd like to load only smaller images if user has mobile browser or hasn't opened his browser in fullscreen.

**Please tell us about your environment:**
Vhecked on ubuntu and iOS.

* **Angular version:** 2.4.1

* **Browser:** [Firefox XX | IE ?? | Safari | iOS ]

### PR DESCRIPTION
**Pull request by DzmitryShylovich**
Closes #13947

### CODE REVIEW

**Comment by mhevery** [comment]
@tbosch please review

File: modules/@angular/core/src/linker/view_utils.ts [hunk 1]
```diff
@@ -425,11 +425,14 @@ function camelCaseToDashCase(input: string): string {
 export function createRenderElement(
     renderer: Renderer, parentElement: any, name: string, attrs: InlineArray<string>,
     debugInfo?: RenderDebugInfo): any {
-  const el = renderer.createElement(parentElement, name, debugInfo);
-  for (let i = 0; i < attrs.length; i += 2) {
-    renderer.setElementAttribute(el, attrs.get(i), attrs.get(i + 1));
+  let attrMap: Map<string, string> = null;
```
Comment (tbosch) [inline modules/@angular/core/src/linker/view_utils.ts]: Can we have a separate method `appendChild`? This way, we don't need to convert the `InlineArray` into a `Map`... We can make `createElement` keep its `parentElement` argument so we don't break users, but we would always call it with `null`.

**Comment by tbosch** [comment]
This seems like a bug in Firefox. E.g. what would happen if the user used attribute binding instead of fixed attributes? E.g. `<source [attr.media]="...">`?

So if we want to make this change, we need at least a test for the `picture` element so that we don't regress later...

**Comment by tbosch** [comment]
This is a breaking change for implementors of `Renderer` (e.g. NativeScript`), so we need to change the commit message to include `BREAKING CHANGE`

**Comment by tbosch** [comment]
@mhevery Can you make the decision whether we should make this change to work around Firefox, and be aware that the workaround will break as soon as users use attribute bindings?

**Comment by DzmitryShylovich** [comment]
> This seems like a bug in Firefox. E.g. what would happen if the user used attribute binding instead of fixed attributes?

yeah, also thought about that. But it doesn't work in Safari as well.

**Comment by DzmitryShylovich** [comment]
@tbosch @mhevery 
I've just tested it https://plnkr.co/edit/rZ6kcsDH33jNxF7dtY3T?p=preview in Mozilla Nightly and it does work as expected.

**Comment by mhevery** [comment]
@IgorMinar please chime in as well.

@DzmitryShylovich we need tests and clear description of breaking change.

**Comment by DzmitryShylovich** [comment]
why do you think it's a breaking change? `attrs` argument is optional so it won't break any existing code

**Comment by mhevery** [comment]
@DzmitryShylovich because the implementor of the interface needs to take that into account. Normally optional args are BC for callers, but they are not for implementors. The direction is reversed.

**Pull request closed without merging**